stdlib/public/stubs/ThreadLocalStorage.cpp

```diff
@@ -54,6 +54,44 @@ _stdlib_thread_key_create(__swift_thread_key_t * _Nonnull key,
 
 #endif
 
+#if defined(__wasi__)
+// WebAssembly doesn't have threading standardized yet, so TLS is emulated
+// here, see https://bugs.swift.org/browse/SR-12774.
+#include <map>
+using __swift_thread_key_destructor = void (*)(void *);
+
+struct _stdlib_tls_element_t {
+  const void *value;
+  __swift_thread_key_destructor destructor;
+};
+
+using _stdlib_tls_map_t = std::map<__swift_thread_key_t, _stdlib_tls_element_t>;
+static void *_stdlib_tls_map;
+
+static inline int _stdlib_thread_key_create(__swift_thread_key_t *key,
+                          __swift_thread_key_destructor destructor) {
+  if (!_stdlib_tls_map)
+      _stdlib_tls_map = new _stdlib_tls_map_t();
```

Comment:
```suggestion
    _stdlib_tls_map = new _stdlib_tls_map_t();
```

Nit: if still pursuing this PR, would you mind running the changes through the clang-format tool?